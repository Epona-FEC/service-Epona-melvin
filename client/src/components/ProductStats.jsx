import React from 'react';
import PropTypes from 'prop-types';
import HalfStar from './HalfStar.jsx';
import EmptyStar from './EmptyStar.jsx';
import FullStar from './FullStar.jsx';


const renderStarRatings = (score) => {
  const stars = [];
  let currentScore = score;
  for (let i = 0; i < 5; i += 1) {
    if (Math.sign(currentScore) === 1 && currentScore >= 1) {
      stars.push(<span><FullStar /></span>);
    } else if (currentScore <= 0) {
      stars.push(<span><EmptyStar /></span>);
    } else if (currentScore < 1 && !Number.isInteger(currentScore - 1) && currentScore > 0) {
      stars.push(<span><HalfStar /></span>);
      currentScore -= currentScore;
    }

    currentScore -= 1;
  }
  return stars;
};

const ProductStats = ({ reviewScore, totalShopReviews }) => (
  <div className="product-stats flex-row">
    <div className="shop-rating-picture flex-column">
      <span>Shop Rating</span>
      <span>{renderStarRatings(reviewScore)}</span>
    </div>
    <div className="shop-rating-container flex-column">
      <span>Total shop reviews</span>
      <span>{totalShopReviews}</span>
    </div>
  </div>
);

ProductStats.propTypes = {
  reviewScore: PropTypes.number,
  totalShopReviews: PropTypes.number,
};

ProductStats.defaultProps = {
  reviewScore: 5,
  totalShopReviews: 323,
};

export default ProductStats;
