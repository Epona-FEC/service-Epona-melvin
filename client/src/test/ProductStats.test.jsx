/* eslint-disable no-undef */
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import ProductStats from '../components/ProductStats.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('ProductStats', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ProductStats />);
  });

  it('Should shallow', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renderStars when function is called, should manipulate reviewScore prop', () => {
    const addOne = (val) => val + 1; // Default reviewScore is 5.
    wrapper.setProps({ renderStars: addOne });
    expect(wrapper.find('.shop-rating-picture').props().children[1].props.children).toBe(6);
  });
});
