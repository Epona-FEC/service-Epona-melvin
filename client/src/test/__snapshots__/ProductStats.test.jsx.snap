// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductStats Should shallow 1`] = `
<div
  className="product-stats flex-row"
>
  <div
    className="shop-rating-picture flex-column"
  >
    <span>
      Shop Rating
    </span>
    <span />
  </div>
  <div
    className="shop-rating-container flex-column"
  >
    <span>
      Total shop reviews
    </span>
    <span>
      323
    </span>
  </div>
</div>
`;
